yocs_cmd_vel_mux/src/cmd_vel_mux_nodelet.cpp

```diff
@@ -29,46 +29,59 @@ namespace yocs_cmd_vel_mux {
 
 void CmdVelMuxNodelet::cmdVelCallback(const geometry_msgs::Twist::ConstPtr& msg, unsigned int idx)
 {
+  // Reset general timer
+  common_timer.stop();
+  common_timer.start();
+
   // Reset timer for this source
-  cmd_vel_sub[idx].timer.stop();
-  cmd_vel_sub[idx].timer.start();
+  cmd_vel_subs[idx]->timer.stop();
+  cmd_vel_subs[idx]->timer.start();
 
-  cmd_vel_sub[idx].active = true;   // obviously his source is sending commands, so active
+  cmd_vel_subs[idx]->active = true;   // obviously his source is sending commands, so active
 
   // Give permit to publish to this source if it's the only active or is
   // already allowed or has higher priority that the currently allowed
-  if ((cmd_vel_sub.allowed == VACANT) ||
-      (cmd_vel_sub.allowed == idx)    ||
-      (cmd_vel_sub[idx].priority > cmd_vel_sub[cmd_vel_sub.allowed].priority))
+  if ((cmd_vel_subs.allowed == VACANT) ||
+      (cmd_vel_subs.allowed == idx)    ||
+      (cmd_vel_subs[idx]->priority > cmd_vel_subs[cmd_vel_subs.allowed]->priority))
   {
-    if (cmd_vel_sub.allowed != idx)
+    if (cmd_vel_subs.allowed != idx)
     {
-      cmd_vel_sub.allowed = idx;
+      cmd_vel_subs.allowed = idx;
 
       // Notify the world that a new cmd_vel source took the control
       std_msgs::StringPtr acv_msg(new std_msgs::String);
-      acv_msg->data = cmd_vel_sub[idx].name;
+      acv_msg->data = cmd_vel_subs[idx]->name;
       active_subscriber.publish(acv_msg);
     }
 
-    mux_cmd_vel_pub.publish(msg);
+    output_topic_pub.publish(msg);
   }
 }
 
 void CmdVelMuxNodelet::timerCallback(const ros::TimerEvent& event, unsigned int idx)
 {
-  if (cmd_vel_sub.allowed == idx)
+  if (cmd_vel_subs.allowed == idx || (idx == 8888 && cmd_vel_subs.allowed != VACANT))
   {
+    if (idx == 8888)
+    {
+      // No cmd_vel messages timeout happened for ANYONE, so last active source got stuck without further
+      // messages; not a big problem, just dislodge it; but possibly reflect a problem in the controller
```

Comment:
Hehe,,  not proud of my magic number neither. I cannot remember why I didn't use constant.

The rational behind the global timer was to ensure that there's no an 'active' input (the one allowed to command the robot) that is publishing nothing (probably because it died). You are right this should not happen with the per-input timers, but... actually it can happen when you tangle with the priorities on runtime. Mmm... this was some time ago, so I cannot recall how!